Rewrite Get Started for the Collector and separate install instructions (#3412)
open-telemetry/opentelemetry.io: content/en/docs/collector/getting-started.md

---
title: Getting Started
# prettier-ignore
cSpell:ignore: otelcol telemetrygen docker jaeger zipkin prometheus dpkg
weight: 1
---

The OpenTelemetry Collector listens to metrics, logs, and traces, processes the
telemetry, and exports it to a wide variety of observability back-ends using its

> components. For a conceptual overview of the Collector, see [Collector][].

Reviewer: ```suggestion
components. For a conceptual overview of the Collector, read the [introduction][].
```

> The following tutorial shows how to deploy the Collector in agent mode and send

Reviewer: ```suggestion
The following tutorial shows how to deploy the Collector and send
```

I think at this place the deployment pattern is not relevant and leads to more confusion

> telemetry to it using the default configuration and the telemetrygen utility.

Reviewer: ```suggestion
telemetry to it using the default configuration and the [telemetrygen](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/cmd/telemetrygen) utility.
```

## Prerequisites

> To follow this tutorial you need the following Go 1.20 or higher.

Reviewer: ```suggestion
To follow this tutorial you need the following

*  Go >= 1.20
```

## Test the OpenTelemetry Collector in five minutes

1. [Download and install](/docs/collector/installation) the Collector for your
   operating system and architecture. For example:

   ```sh
   sudo apt-get update
   sudo apt-get -y install wget systemctl
   wget https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v{{% param collectorVersion %}}/otelcol_{{% param collectorVersion %}}_linux_amd64.deb
   sudo dpkg -i otelcol_{{% param collectorVersion %}}_linux_amd64.deb
   ```

   Alternatively, you can download and run the Docker container:

   ```sh
   docker pull otel/opentelemetry-collector-contrib:{{% param collectorVersion %}}
   docker run otel/opentelemetry-collector-contrib:{{% param collectorVersion %}}

>    ```

Reviewer: Let's keep it simple here and tell people that for the sake of the getting started they should use docker and they will find different deployment patterns on that other page.

Note: let's not use contrib here, I don't think you will make use of any additional feature of it?

Reviewer: Also, add some notes on what they should see on their console

Author: Solved in https://github.com/open-telemetry/opentelemetry.io/pull/3412/commits/5826d97fc3f8b83911360fa7fa850fa91f1576b1

Note that at least two ports must be exposed when running the Docker container

2. Download and install the telemetrygen utility from the
   [opentelemetry-collector-contrib] repository:

   ```sh
   go install github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen@latest
   ```

3. The telemetrygen command generates dummy telemetry for testing. Try sending
   some traces to the Collector:

   ```sh
   telemetrygen traces --otlp-insecure --duration 5s
   ```

4. Open http://localhost:55679/debug/tracez in your browser and select one of
   the samples in the table to see the traces you've just generated.

## Collector demo

The OpenTelemetry Collector demo deploys a load generator, agent, and gateway,
as well as Jaeger, Zipkin and Prometheus back-ends.

To try the demo, run the following commands:

```sh
git clone git@github.com:open-telemetry/opentelemetry-collector-contrib.git --depth 1; \
  cd opentelemetry-collector-contrib/examples/demo; \
  docker compose up -d
```

> You can find more information about the demo in the [README.md][] file.

Reviewer: Let's remove that section about the demo

Author: Done in https://github.com/open-telemetry/opentelemetry.io/pull/3412/commits/c1b457308a14b07867b4f414b2ba5ff67cdbb541
